packages/user-interface/src/components/Document.tsx

```diff
@@ -2,42 +2,18 @@ import { useContext } from "react";
 import prettyBytes from "pretty-bytes";
 import { LocaleContext } from "@nl-portal/nl-portal-localization";
 import { Document as PortalDocument } from "@nl-portal/nl-portal-api";
-import { KeycloakContext } from "@nl-portal/nl-portal-authentication";
 import { File } from "@gemeente-denhaag/file";
+import useDownload from "../hooks/useDownload";
 
 interface Props {
   document: PortalDocument;
   downloadLink: string;
 }
 
 const Document = ({ document: doc, downloadLink: downloadLink }: Props) => {
-  const { keycloakToken } = useContext(KeycloakContext);
+  const { onClick } = useDownload(downloadLink, doc.bestandsnaam || "");
```

Comment:
Is de bestandsnaam or hier nodig als hij optioneel is?

Reply:
Aangepast!